extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js

```diff
@@ -246,40 +315,51 @@ class LifecycleBucketProcessor {
     _setupProducer(cb) {
         const producer = new BackbeatProducer({
             kafka: { hosts: this._kafkaConfig.hosts },
+            topic: this._lcConfig.objectTasksTopic,
+        });
+        producer.once('error', err => {
+            this._log.error('error setting up kafka producer', {
+                error: err,
+                method: 'LifecycleBucketProcesso::_setupProducer',
+            });
+            process.exit(1);
         });
-        producer.once('error', cb);
         producer.once('ready', () => {
+            this._log.debug('producer is ready',
+                { kafkaConfig: this.kafkaConfig });
             producer.removeAllListeners('error');
             producer.on('error', err => {
                 this._log.error('error from backbeat producer', {
                     error: err,
                 });
             });
-            return cb(null, producer);
+            this._producerReady = true;
+            this._producer = producer;
+            return cb();
         });
     }
 
     /**
      * Set up the lifecycle consumer.
+     * @param {function} cb - callback
      * @return {undefined}
      */
-    _setupConsumer() {
-        let consumerReady = false;
+    _setupConsumer(cb) {
         this._consumer = new BackbeatConsumer({
             zookeeper: {
                 connectionString: this._zkConfig.connectionString,
             },
             kafka: {
                 hosts: this._kafkaConfig.hosts,
-                backlogMetrics: this._kafkaConfig.backlogMetrics,
+                // backlogMetrics: this._kafkaConfig.backlogMetrics,
```

Comment:
Is this not expected to work?

Reply:
`BackbeatConsumer` at the moment does not have the `backlogMetrics` implemented, and, in my opinion, is out of the scope of this ticket